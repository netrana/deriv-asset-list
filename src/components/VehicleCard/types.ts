export interface Props {
    title: string;
    cardImage?: string;
    rating?: number;
    tripCount?: number;
    perDayCost?: number;
    isNew?: boolean;
    className?: string;
    onClick?:()=>void;
  }