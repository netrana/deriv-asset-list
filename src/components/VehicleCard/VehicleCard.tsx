import { StarFilled } from '@ant-design/icons';
import { Col, Row } from 'antd';
import React, { FC } from 'react';

import NXCard from 'components/NXCard';

import { Props } from './types';
import styles from './vehicleCard.module.scss';

const VehicleCard: FC<Props> = (props) => {
  const {
    title,
    cardImage,
    rating,
    tripCount,
    perDayCost = 0,
    isNew = true,
    onClick,
  } = props;
  return (
    <NXCard
      cssClassNames={styles.vehicleCard}
      isBordered={true}
      image={cardImage}
      isHoverable={true}
      metaTitle={title}
      metaDescription={
        <>
          {(rating || rating === 0) && (
            <StarFilled className={styles.starIcon} />
          )}{' '}
          <b>{rating}</b>{' '}
          {tripCount && tripCount !== 0 && (
            <span className={styles.trips}>({tripCount} trips)</span>
          )}
        </>
      }
      footer={
        <>
          <Row className={styles.footerBody}>
            <Col span={16} className={styles.footerLeft}>
              <span> ${perDayCost * 1}/day</span>
            </Col>
            <Col span={8} className={styles.footerRight}>
              {isNew && <span className={styles.statusButton}>New</span>}
            </Col>
          </Row>
        </>
      }
      onClick={onClick}
    />
  );
};

export default VehicleCard;
