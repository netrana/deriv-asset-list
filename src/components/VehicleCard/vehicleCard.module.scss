@import 'styles/globals.scss';

.vehicleCard {
  margin: 1rem 0 0 0;
  .starIcon {
    color: $primaryThemeColor;
  }

  .footerBody {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 1rem;
    color: rgba(0, 0, 0, 0.85);
    font-weight: 600;
    font-size: 16px;
    line-height: 1.5;
    @media screen and (max-width: 600px) {
      font-size: 0.8rem;
    }
  }

  .footerLeft {
    text-align: left;
  }

  .footerRight {
    text-align: right;
  }

  .statusButton {
    border: 1px solid $primaryThemeColor;
    color: $primaryThemeColor;
    text-align: center;
    border-radius: 2px;
    padding: 0 1rem;

    @media screen and (max-width: 600px) {
      padding: 0 0.8rem;
    }
  }
}
