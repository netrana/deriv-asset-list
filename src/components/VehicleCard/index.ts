import VehicleCard from "./VehicleCard";

export default VehicleCard;