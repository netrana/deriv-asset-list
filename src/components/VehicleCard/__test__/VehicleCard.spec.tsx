import React from "react";
import render from "react-test-renderer";
import VehicleCard from "../VehicleCard";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Card: "Card",
}));

jest.mock("@ant-design/icons", () => ({
  ...jest.requireActual("@ant-design/icons"),
  RightOutlined: "RightOutlined",
  StarFilled: "StarFilled",
}));

describe("VehicleCard component test", () => {
  describe("snapshot test", () => {
    it("should match snapshot with default props", () => {
      const tree = render
        .create(<VehicleCard primaryTitle="Aston Martin" />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should match snapshot for small card", function () {
      const tree = render
        .create(
          <VehicleCard primaryTitle="Aston Martin" cardType="smallCard" />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should match snapshot for destination card", function () {
      const tree = render
        .create(
          <VehicleCard primaryTitle="Aston Martin" cardType="destinationCard" />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
