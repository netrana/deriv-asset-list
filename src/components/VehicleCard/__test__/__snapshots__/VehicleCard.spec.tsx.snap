// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard component test snapshot test should match snapshot for destination card 1`] = `
<div
  className="container vehicleCard"
>
  <Card
    actions={
      Array [
        <React.Fragment>
          <Row
            className="footerBody"
          >
            <Col
              className="footerLeft"
              span={16}
            >
              <span>
                 $
                0
                /day
              </span>
            </Col>
            <Col
              className="footerRight"
              span={8}
            >
              <span
                className="statusButton"
              >
                New
              </span>
            </Col>
          </Row>
        </React.Fragment>,
      ]
    }
    bordered={false}
    className="hasBorder"
    cover={null}
    hoverable={true}
    loading={false}
  />
</div>
`;

exports[`VehicleCard component test snapshot test should match snapshot for small card 1`] = `
<div
  className="container vehicleCard"
>
  <Card
    actions={
      Array [
        <React.Fragment>
          <Row
            className="footerBody"
          >
            <Col
              className="footerLeft"
              span={16}
            >
              <span>
                 $
                0
                /day
              </span>
            </Col>
            <Col
              className="footerRight"
              span={8}
            >
              <span
                className="statusButton"
              >
                New
              </span>
            </Col>
          </Row>
        </React.Fragment>,
      ]
    }
    bordered={false}
    className="hasBorder"
    cover={null}
    hoverable={true}
    loading={false}
  />
</div>
`;

exports[`VehicleCard component test snapshot test should match snapshot with default props 1`] = `
<div
  className="container vehicleCard"
>
  <Card
    actions={
      Array [
        <React.Fragment>
          <Row
            className="footerBody"
          >
            <Col
              className="footerLeft"
              span={16}
            >
              <span>
                 $
                0
                /day
              </span>
            </Col>
            <Col
              className="footerRight"
              span={8}
            >
              <span
                className="statusButton"
              >
                New
              </span>
            </Col>
          </Row>
        </React.Fragment>,
      ]
    }
    bordered={false}
    className="hasBorder"
    cover={null}
    hoverable={true}
    loading={false}
  />
</div>
`;
